packages/teams-js/src/public/dialog.ts

```diff
@@ -210,6 +217,121 @@ export namespace dialog {
+      width: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: botAdaptiveCardDialogInfo.title,
+      completionBotId: botAdaptiveCardDialogInfo.completionBotId,
+    };
+    return dialogInfo;
+  }
+  /**
+   * Subcapability for interacting with adaptive card dialogs
+   */
+  export namespace adaptiveCard {
+    /**
+     * Allows app to open an adaptive card based dialog.
+     *
+     * @remarks
+     * This function cannot be called from inside of a dialog
+     *
+     * @param adaptiveCardDialogInfo - An object containing the parameters of the dialog module.
+     * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode submit} function or when the user closes the dialog.
+     */
+    export function open(adaptiveCardDialogInfo: AdaptiveCardDialogInfo, submitHandler?: DialogSubmitHandler): void {
+      ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      const dialogInfo: DialogInfo = getDialogInfoFromAdaptiveCardDialogInfo(adaptiveCardDialogInfo);
+      sendMessageToParent('tasks.startTask', [dialogInfo], (err: string, result: string | object) => {
+        submitHandler?.({ err, result });
+        removeHandler('messageForParent');
+      });
+    }
+
+    /**
+     * Checks if dialog.adaptiveCard module is supported by the host
+     *
+     * @returns boolean to represent whether dialog.adaptiveCard module is supported
+     */
+    export function isSupported(): boolean {
+      return runtime.supports.dialog && runtime.supports.dialog.adaptiveCard ? true : false;
+    }
+
+    /**
+     * Namespace for interaction with adaptive card dialogs that need to communicate with the bot framework
+     */
+    export namespace bot {
+      /**
+       * Allows an app to open a adaptive card-based dialog module using bot.
+       *
+       * @param botAdaptiveCardDialogInfo - An object containing the parameters of the dialog module including completionBotId.
+       * @param submitHandler - Handler that triggers when the dialog has been submitted or closed.
+       *
+       * @returns a function that can be used to send messages to the dialog.
+       */
+      export function open(
+        botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+        submitHandler?: DialogSubmitHandler,
+      ): void {
+        ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
+        if (!isSupported()) {
+          throw errorNotSupportedOnPlatform;
+        }
+
+        const dialogInfo: DialogInfo = getDialogInfoFromBotAdaptiveCardDialogInfo(botAdaptiveCardDialogInfo);
+
+        sendMessageToParent('tasks.startTask', [dialogInfo], (err: string, result: string | object) => {
+          submitHandler?.({ err, result });
```

Comment:
why does this not have removeHandler call?

Reply:
There is no handler registered in the code.